src/odbc/src/app/application_data_buffer.cpp

```diff
@@ -282,7 +282,7 @@ ConversionResult::Type ApplicationDataBuffer::PutStrToStrBuffer(
   SqlUlen inCharIndex = currentCellOffset / inCharSize;
 
   if (inCharIndex >= value.length()) {
-    if (resLenPtr) {
+    if (resLenPtr && value.length() != 0) {
       *resLenPtr = SQL_NO_TOTAL;
```

Comment:
I still feel like this is wrong...?
SQL_NO_TOTAL should only ever be returned if there is more data in the cell (ie it got truncated) but this looks like it occurs if we are calling SQLGetData past the end of the cell.

Reply:
Really, Line 335 is the scenario where SQL_NO_TOTAL should be returned, but the remaining length is already calculated so it's not necessary.

Reply:
Yes, this looks like more data in the cell. It is not the return size unknown.